task --slient --list-all with remote taskfiles causes panic

A simple repro case:

```
# Taskfile.yml
version: "3"

includes:
  scary: https://raw.githubusercontent.com/go-task/task/main/docs/Taskfile.yml
```

Pull the latest version of task:

```
$ mkdir task3.34.1
$ curl -Ls https://github.com/go-task/task/releases/download/v3.34.1/task_darwin_arm64.tar.gz | tar -xvzf - -C task3.34.1
```

Run the following

```
$ ./task3.33.1/task --silent --list-all
# OR
$ ./task3.33.1/task --yes --list-all
$ ./task3.33.1/task --silent --list-all
```

See the error

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=...]
...
```

This may be causing issues for #1406 since the [bash auto complete uses `--silent --list-all`](https://github.com/go-task/task/blob/1f477eb456c60e96ddd2cc41da8a7d3a28d78a9d/completion/bash/task.bash#L48)


Task version: 3.34.1
Operating system: macOS 14.2
Experiments enabled: `TASK_X_REMOTE_TASKFILES=1`

--- a/cmd/task/task.go
+++ b/cmd/task/task.go
@@ -265,8 +265,7 @@ func run() error {
 	}
 
 	if (listOptions.ShouldListTasks()) && flags.silent {
-		e.ListTaskNames(flags.listAll)
-		return nil
+		return e.ListTaskNames(flags.listAll)
 	}
 
 	if err := e.Setup(); err != nil {

--- a/help.go
+++ b/help.go
@@ -122,6 +122,10 @@ func (e *Executor) ListTasks(o ListOptions) (bool, error) {
 // Only tasks with a non-empty description are printed if allTasks is false.
 // Otherwise, all task names are printed.
 func (e *Executor) ListTaskNames(allTasks bool) {
+	// if called from cmd/task.go, e.Logger has not yet been initialized
+	if e.Logger == nil {
+		e.setupLogger()
+	}
 	// if called from cmd/task.go, e.Taskfile has not yet been parsed
 	if e.Taskfile == nil {
 		if err := e.readTaskfile(); err != nil {

--- a/help.go
+++ b/help.go
@@ -5,7 +5,6 @@ import (
 	"encoding/json"
 	"fmt"
 	"io"
-	"log"
 	"os"
 	"strings"
 	"text/tabwriter"
@@ -121,7 +120,7 @@ func (e *Executor) ListTasks(o ListOptions) (bool, error) {
 // ListTaskNames prints only the task names in a Taskfile.
 // Only tasks with a non-empty description are printed if allTasks is false.
 // Otherwise, all task names are printed.
-func (e *Executor) ListTaskNames(allTasks bool) {
+func (e *Executor) ListTaskNames(allTasks bool) error {
 	// if called from cmd/task.go, e.Logger has not yet been initialized
 	if e.Logger == nil {
 		e.setupLogger()
@@ -129,8 +128,7 @@ func (e *Executor) ListTaskNames(allTasks bool) {
 	// if called from cmd/task.go, e.Taskfile has not yet been parsed
 	if e.Taskfile == nil {
 		if err := e.readTaskfile(); err != nil {
-			log.Fatal(err)
-			return
+			return err
 		}
 	}
 	// use stdout if no output defined
@@ -161,6 +159,7 @@ func (e *Executor) ListTaskNames(allTasks bool) {
 	for _, t := range taskNames {
 		fmt.Fprintln(w, t)
 	}
+	return nil
 }
 
 func (e *Executor) ToEditorOutput(tasks []*ast.Task, noStatus bool) (*editors.Taskfile, error) {

